geomesa-gs-styling/src/main/scala/org/geomesa/gs/styling/filter/GeoMesaLabelParser.scala

```diff
@@ -87,14 +88,15 @@ class GeoMesaLabelParser4 extends FunctionExpressionImpl(
 }
 
 object GeoMesaLabelParser {
+  val p = Pattern.compile("\\d*\\.\\d*")
   def formatProp (expr: Any, format: String): String = {
     if (expr == null) {
       ""
     } else {
       // expr could be an Int/Float/Double
       val prop = expr.toString
-      // If there is a letter in the prop value, we can't parse so short circuit
-      if (prop.matches(".*[a-zA-Z].*") || prop == "") {
+      // Only attempt parse if there is a decimal point
+      if (!p.matcher(prop).matches || prop == "") {
```

Comment:
the empty string check is redundant now, no? could maybe move it first if it would be faster, otherwise remove it